pkg/apis/kudo/v1beta1/instance_types_helpers.go

```diff
@@ -233,6 +233,24 @@ func GetParamDefinitions(params map[string]string, ov *OperatorVersion) []Parame
 	return defs
 }
 
+// GetParamDefinitions retrieves parameter metadata from OperatorVersion but returns an error if any parameter is missing
+func GetParamDefinitions(params map[string]string, ov *OperatorVersion) ([]Parameter, error) {
+	defs := []Parameter{}
+	for p1 := range params {
+		p1 := p1
+		p2 := funk.Find(ov.Spec.Parameters, func(e Parameter) bool {
```

Comment:
This is `O(N^2)` AFAICT. Can you please copy `ov.Spec.Parameters` into a map instead and compute and error on parameters not in the intersection?

Reply:
I'm not sure it's an improvement. Giving that we don't have thousands of parameters and the fact that traversing arrays is [faster than map lookup](https://go-talks.appspot.com/github.com/dgryski/talks/dotgo-2016/slices.slide#12) (see slide 12) on small `n` I don't think it's needed here.

Reply:
Well, some operators have (admittedly low) hundreds of them :-)
I'm aware that simple algorithms are usually faster for small inputs, but I'm a little surprised that maps only get faster around 30-ish elements in Go. I was expecting 10 or so.
Also, I don't think we really are in position to tell whether this is a performance problem or not, because we do not have performance tests :see_no_evil: 
But if there is consensus that this might be premature optimization, then I'm not going to block this!